pkg/operator/ceph/nfs/spec.go

```diff
@@ -128,7 +133,7 @@ func (r *ReconcileCephNFS) makeDeployment(nfs *cephv1.CephNFS, cfg daemonConfig)
 			nfsConfigVol,
 			dbusVol,
 		},
-		HostNetwork:       r.cephClusterSpec.Network.IsHost(),
+		HostNetwork:       hostNetwork,
```

Comment:
@travisn when we use host networking for the Ceph cluster, we also skip creating the Services right? Shoudln't we also remove any services if hostNetwork is enabled in NFS?

Reply:
My $0.02 would be that it should still create the service simply because the service could be used to target something -- even if it's using host networking it might be useful to access it by name using dns.

As I was looking at that, though, I found two more places in the file that I should have updated but didn't around DNS, so I'll fix those and push the changes soon.

Reply:
Actually, earlier in that file there is a check that makes it a headless service if Host networking is enabled -- that seems like the right thing to do to me. I've updated that check to use the new logic as well as two other places in the file that I found and pushed those changes.

Reply:
Makes sense to me. I'd suggest to go ahead and squash them, thanks.

Reply:
squashed

Reply:
Are you able to verify that the service correctly routes to the NFS if it's in host network mode? As long as that works, I think this is good to go.

Reply:
I don't have a way to actually test it, but I can tell you that a headless service for a pod using host networking will resolve to the ip address(es) of the host(s) indicated -- that's easy enough to test using other similar services that I already have running.

Additionally there are no changes to what it does when in host network mode, only in how it decides if it is in host network mode or not.

Reply:
Is there anything else you'd like me to do on this?